### PR TITLE
fix: empty message bug

### LINKED ISSUE #356: [BUG] don't allow empty quote
### Description

if a quote reply is empty, a message shouldn't be sent

### Screenshots

_No response_

### Additional information

_No response_

### 🥦 Browser

Google Chrome

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

None

### PR DESCRIPTION
**Pull request by shubho0908**
fixes #356 

My PR closes #issue_number_here

# 👨‍💻 Changes proposed(What did you do ?)

Made changes to the handleSubmit function to ensure that when you're replying to a quote, an empty message won't be sent. That's it!

# ✔️ Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

image

### CODE REVIEW

**Comment by Dun-sin** [comment]
> # PR Compliance Checks
> 
> Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> ## Issue Reference In order to be considered for merging, the pull request description must refer to a specific issue number. This is described in our [Contributing Guide](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md). This check is looking for a phrase similar to: "Fixes #XYZ" or "Resolves #XYZ" where XYZ is the issue number that this PR is meant to address.
> ## Conventional Commit PR Title
> 
> In order to be considered for merging, the pull request title must match the specification in [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). You can edit the title in order for this check to pass. Most often, our PR titles are something like one of these:
> 
>     * docs: correct typo in README
> 
>     * feat: implement dark mode"
> 
>     * fix: correct remove button behavior
> 
> 
> Linting Errors
> 
>     * Found type "null", must be one of "feat","fix","docs","style","refactor","perf","test","build","ci","chore","revert"
> 
>     * No subject found

@shubho0908 
Please fix this error, if you don't know how, try this:
- Read the error carefully or read the contributing file
- Checkout past merged Pull requests to see how they were done for it to be merged

> Note: you can edit your PR message, this PR won't be merged if the error isn't gone and you have a week max to fix it before the PR gets closed without being merged

**Comment by shubho0908** [comment]
> > # PR Compliance Checks
> > Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> > ## Issue Reference In order to be considered for merging, the pull request description must refer to a specific issue number. This is described in our [Contributing Guide](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md). This check is looking for a phrase similar to: "Fixes #XYZ" or "Resolves #XYZ" where XYZ is the issue number that this PR is meant to address.
> > ## Conventional Commit PR Title
> > In order to be considered for merging, the pull request title must match the specification in [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). You can edit the title in order for this check to pass. Most often, our PR titles are something like one of these:
> > ```
> > * docs: correct typo in README
> > 
> > * feat: implement dark mode"
> > 
> > * fix: correct remove button behavior
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Linting Errors
> > ```
> > * Found type "null", must be one of "feat","fix","docs","style","refactor","perf","test","build","ci","chore","revert"
> > 
> > * No subject found
> > ```
> 
> @shubho0908 Please fix this error, if you don't know how, try this:
> 
> * Read the error carefully or read the contributing file
> * Checkout past merged Pull requests to see how they were done for it to be merged
> 
> > Note: you can edit your PR message, this PR won't be merged if the error isn't gone and you have a week max to fix it before the PR gets closed without being merged

Fixed the PR errors, please review the changes.

File: client/src/components/Chat.jsx [hunk 1]
```diff
@@ -213,60 +213,56 @@ const Chat = () => {
...
-        let message = inputRef.current.value.trim();        // Trim the message to remove the extra spaces
-
-        if (!isQuoteReply) {
-            const cleanedText = message.replace(/>+/g, '');
-            message = cleanedText
-        }
-
-        if (message === '' || senderId === undefined || senderId === '123456') {
+    
+        const currentTime = new Date().getTime();
```
Comment (Dun-sin) [inline client/src/components/Chat.jsx:220]: Why did you change this?

File: client/src/components/Chat.jsx [hunk 2]
```diff
@@ -213,60 +213,56 @@ const Chat = () => {
...
+    
+        let message = inputRef.current.value.trim();
+    
+        if (isQuoteReply && message === '') {
```
Comment (Dun-sin) [inline client/src/components/Chat.jsx:224]: What about the old condition, you can't just remove it

File: client/src/components/Chat.jsx (hunk 1 above)
Comment (shubho0908) [reply client/src/components/Chat.jsx:220]: if (message === '' || senderId === undefined || senderId === '123456') {
    return;
}

This condition checked if the message was empty or if the senderId was undefined or equal to '123456'. it did not consider the case where the message might consist of only spaces or tabs, allowing empty messages with whitespace to be sent.

Comment (Dun-sin) [reply client/src/components/Chat.jsx:220]: That's not what I'm talking about, I'm talking about the current time

**Comment by darknight2135** [comment]
is this still open? can I try?
